
From: Ronald Bultje <rbultje@ronald.bitfreak.net>

attached patch makes the zr36067 driver use msleep() instead of
schedule_timeout() with uninterruptible state. Patch originally
submitted by Nishanth Aravamudan <nacc@us.ibm.com> (7/26).

Signed-off-by: Ronald Bultje <rbultje@ronald.bitfreak.net>
Signed-off-by: Nishanth Aravamudan <nacc@us.ibm.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/media/video/zoran_device.c |    3 +--
 1 files changed, 1 insertion(+), 2 deletions(-)

diff -puN drivers/media/video/zoran_device.c~zr36067-driver-use-msleep-instead-of-schedule_timeout drivers/media/video/zoran_device.c
--- 25/drivers/media/video/zoran_device.c~zr36067-driver-use-msleep-instead-of-schedule_timeout	2004-08-30 00:03:47.732213168 -0700
+++ 25-akpm/drivers/media/video/zoran_device.c	2004-08-30 00:03:47.738212256 -0700
@@ -1105,8 +1105,7 @@ zr36057_enable_jpg (struct zoran        
 			ZR36057_ISR);
 		btand(~ZR36057_JMC_Go_en, ZR36057_JMC);	// \Go_en
 
-		current->state = TASK_UNINTERRUPTIBLE;
-		schedule_timeout(HZ / 20);
+		msleep(50);
 
 		set_videobus_dir(zr, 0);
 		set_frame(zr, 1);	// /FRAME
_
